src/core/string_map.cc

```diff
@@ -101,6 +101,36 @@ sds StringMap::Find(std::string_view key) {
   return GetValue(str);
 }
 
+sds StringMap::ReallocIfNeeded(void* obj, float ratio) {
+  sds key = (sds)obj;
```

Comment:
What about the actual `DenseSet` contents? a.k.a the keys (+metadata)?

Reply:
they key is re-alloced below, metadata is a single vector (not touched) as well as link entries